docs/application-tutorial/game_of_life.py

```diff
@@ -0,0 +1,511 @@
+import sys
+import time
+import math
+import numpy as np
+
+# Hide pygame prompt
+from os import environ
+
+environ["PYGAME_HIDE_SUPPORT_PROMPT"] = "1"
+
+import pygame
+import argparse
+
+from concrete import fhe
+
+# Constants of the game of life
+weights_method_3b_a = np.array([[1, 1, 1], [1, 0, 1], [1, 1, 0]])
+weights_method_3b_b = np.array([[0, 0, 0], [0, 0, 0], [0, 0, 1]])
+table_next_cell_3b_a = [i if i <= 3 else 4 for i in range(8)]
+table_next_cell_3b_b = [i - 1 if i in [2, 3] else 0 for i in range(6)]
+table_next_cell_3b_c = [i in [2, 3] for i in range(4)]
+
+weights_method_4b = np.array([[1, 1, 1], [1, 0, 1], [1, 1, 1]])
+table_next_cell_4b_a = [i - 1 if i in [2, 3] else 0 for i in range(9)]
+table_next_cell_4b_b = [i in [2, 3] for i in range(4)]
+
+weights_method_5b = np.array([[1, 1, 1], [1, 9, 1], [1, 1, 1]])
+table_next_cell_5b = [i in [3, 9 + 2, 9 + 3] for i in range(18)]
+
+weights_method_basic = np.array([[1, 1, 1], [1, 0, 1], [1, 1, 1]])
+table_next_cell_basic_a = [i in [3] for i in range(9)]
+table_next_cell_basic_b = [i in [2, 3] for i in range(9)]
+
+
+# CP tables
+table_cp_next_cell_3b_a = fhe.LookupTable(list(map(int, table_next_cell_3b_a)))
+table_cp_next_cell_3b_b = fhe.LookupTable(list(map(int, table_next_cell_3b_b)))
+table_cp_next_cell_3b_c = fhe.LookupTable(list(map(int, table_next_cell_3b_c)))
+
+table_cp_next_cell_4b_a = fhe.LookupTable(list(map(int, table_next_cell_4b_a)))
+table_cp_next_cell_4b_b = fhe.LookupTable(list(map(int, table_next_cell_4b_b)))
```

Comment:
You can (and should) directly pass numpy arrays as we have this code in Concrete:
```python
    def __init__(self, table: Any):
        is_valid = True
        try:
            self.table = table if isinstance(table, np.ndarray) else np.array(table)
        except Exception:  # pragma: no cover  # pylint: disable=broad-except
            # here we try our best to convert the table to np.ndarray
            # if it fails we raise the exception at the end of the function
            is_valid = False
```

Reply:
@umut-sahin , it fails for this case:

ValueError: LookupTable cannot be constructed with [False, False, True, True]

which is `table_cp_next_cell_3b_c = fhe.LookupTable(table_next_cell_3b_c)`, which is table_next_cell_3b_c = [i in [2, 3] for i in range(4)]. Do you need to update a bit your __init__? I guess I can astype the input as well, but maybe better it's directly handled in the constructor

Reply:
I have updated the one I can and let a FIXME

Reply:
so, I wait for your advices here, partially done

Reply:
You can do `[int(i in [2, 3]) for i in range(4)]` maybe.

Reply:
Thx, done!